Integrate new navigation

AlistairGempf opened the pull request:
**Overall change:**
There are a number of issues that need to be resolved in order to be able to release the new navigation. These are:

- Integrate Psammead components and handle state for AMP and Canonical. (#4571 and #4570)
- Add entry and exit animation for navigation dropdown, for AMP and Canonical. (#4579)
- Move skip to content link from navigation to brand component (#2829)
- Use script link into the brand component (#4009)

We have created a feature branch so we can review these changes incrementally, rather than having one large review when we are ready for these to go live.

These pull requests handle aspects of this:

**In Progress**
Integrate canonical navigation animation #5057

Then:
 - Incorporate changes from Remove aria-hidden from scrollable navigation (https://github.com/bbc/psammead/pull/2875)
 - Modify e2es
 - Investigate https://github.com/bbc/simorgh/issues/5038
 - Fix script link padding on small devices: https://github.com/bbc/simorgh/issues/5062

**Done**
Navigation: add unit tests and refactor AMP constants #4968
Update nav translations #4969
Spread Navigation props #2802
Add a comment about the hidden attribute #4952
Ensure Nav is always red on desktop #2810
Move media query hooks to Simorgh #2801
Use Brand with Script Link component on services with variants #4948

As well as these issues, which were completed before we decided to use a feature branch:
Navigation: Canonical dropdown integration and dropdown toggling #4571
Navigation: AMP dropdown integration and dropdown toggling #4570

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [x] This PR requires manual testing - we will require cross browser testing across browser matrix

AlistairGempf committed:
First pass at integrating scrollable nav - broken

AlistairGempf committed:
Update to working nav package

AlistairGempf committed:
Split into Amp and canonical scrolling nav

OlgaLyubin committed:
Import dropdown navigation

OlgaLyubin committed:
Resolve conflict

Commit:
Add Canonical menu button handler

Commit:
Trial add of amp handler.

OlgaLyubin committed:
Merge branch 'latest' into integrate-new-navigation

OlgaLyubin committed:
Pull in new version of menu buttons

OlgaLyubin committed:
Resolve conflicts

OlgaLyubin committed:
Update stories to have canonical and amp versions

OlgaLyubin committed:
Fix storybook warnings

OlgaLyubin committed:
Use announcedText for a menu

OlgaLyubin committed:
Pull in new navigation version

OlgaLyubin committed:
Pass scripts so that new button dimensions work

OlgaLyubin committed:
Merge branch 'latest' into integrate-new-navigation

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into integrate-new-navigation

Commit:
Add isOpen prop to <Navigation />. Add script prop to <CanonicalMenuButton />.

Commit:
Merge branch 'integrate-new-navigation' of github.com:bbc/simorgh into integrate-new-navigation

AlistairGempf committed:
Pull in latest version of psammead navigation

AlistairGempf committed:
Update Header test to mock window.matchMedia

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into integrate-new-navigation

AlistairGempf committed:
Test with and without hamburger menu

AlistairGempf committed:
Snapshots

AlistairGempf committed:
Add skip link to Brand tests

AlistairGempf committed:
Update tests that didn't work as hoped

OlgaLyubin committed:
Merge branch 'latest' into integrate-new-navigation

OlgaLyubin committed:
Pull in new nav version

Commit:
Merge branch 'integrate-new-navigation' of github.com:bbc/simorgh into integrate-new-navigation

AlistairGempf committed:
Remove unnecessary props

AlistairGempf committed:
Merge branch 'integrate-new-navigation' of https://github.com/bbc/simorgh into integrate-new-navigation

AlistairGempf committed:
Revert "Remove unnecessary props"

This reverts commit ce1d7047b3674b3433eef5049e18b2341bb5aeb7.

AlistairGempf committed:
Add dir prop to scrolling link items

OlgaLyubin committed:
Fix failing snapshots

OlgaLyubin committed:
Merge branch 'integrate-new-navigation' of github.com:bbc/simorgh into integrate-new-navigation

OlgaLyubin committed:
Merge branch 'latest' into integrate-new-navigation

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into integrate-new-navigation

Commit:
Toggle amp dropdown and scrollable navigation.

OlgaLyubin committed:
Use translated annouced text for sections

OlgaLyubin committed:
Merge branch 'latest' into integrate-new-navigation

Commit:
Merge branch 'integrate-new-navigation' of github.com:bbc/simorgh into integrate-new-navigation

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into integrate-new-navigation

Commit:
Update variable names.

AlistairGempf committed:
Update navigation package to use expected amp toggling

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into integrate-new-navigation

OlgaLyubin committed:
Pull in new navigation version

OlgaLyubin committed:
Fix menu button props

OlgaLyubin committed:
Merge branch 'latest' into integrate-new-navigation

AlistairGempf committed:
Add useMediaQuery hook

AlistairGempf committed:
Update navigation package

AlistairGempf committed:
Integrate media query hook

AlistairGempf committed:
Display nav bar when window resize and dropdown open

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into integrate-new-navigation

AlistairGempf committed:
Use imported screenwidth

Commit:
Merge branch 'integrate-new-navigation' of github.com:bbc/simorgh into integrate-new-navigation

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into integrate-new-navigation

OlgaLyubin committed:
Merge branch 'latest' into integrate-new-navigation

Commit:
Add variants to config.

Commit:
Update psammead-brand version.

AlistairGempf committed:
Import breakpoint for use media query

AlistairGempf committed:
Update package to latest nav

Commit:
Add ScriptLink and Skiplink to Brand. Add variantOffscreenText to config.

Commit:
Update snapshots.

Commit:
Merge branch 'integrate-new-navigation' of github.com:bbc/simorgh into 4009-use-brand-with-script-link

Commit:
Add test.

AlistairGempf commented on src/app/containers/Navigation/index.amp.jsx:
```diff
@@ -0,0 +1,70 @@
+import { AmpScrollableNavigation } from '@bbc/psammead-navigation/scrollable';
+import { AmpMenuButton } from '@bbc/psammead-navigation/dropdown';
+import { GEL_GROUP_2_SCREEN_WIDTH_MAX } from '@bbc/gel-foundations/breakpoints';
+import styled from '@emotion/styled-base';
```
```suggestion
import styled from 'styled-components';
```

AlistairGempf committed:
Update import to styled-components

tochwill commented on src/app/containers/Navigation/index.jsx:
```diff
@@ -1,49 +1,104 @@
-  const { script, translations, navigation, service, dir } = useContext(
-    ServiceContext,
-  );
+const Navigation = () => {
```
Call this NavigationContainer for consistency

tochwill commented on src/app/containers/Navigation/index.jsx:
```diff
@@ -1,49 +1,104 @@
+
+  const dropdownId = 'dropdown-menu';
+  const dropdownListItems = (
+    <Dropdown id={dropdownId} hidden={isAmp}>
```
This ID is added for both canonical and amp. Do we think this is a problem?

tochwill replied:
Also add a comment to explain what `hidden={isAmp}` is doing.

tochwill replied:
We can remove these and add them in the AMP component (have added to PR)

tochwill commented on src/app/containers/Navigation/index.jsx:
```diff
@@ -1,49 +1,104 @@
+  const dropdownId = 'dropdown-menu';
+  const dropdownListItems = (
+    <Dropdown id={dropdownId} hidden={isAmp}>
+      <DropdownUl>
```
Look at refactoring creating the lists into a shared function for both navigations.

tochwill commented on src/app/containers/Navigation/index.jsx:
```diff
@@ -1,49 +1,104 @@
+  );
+
+  return isAmp ? (
+    <Amp
```
Also refactor this to choose the component required rather than use a ternary.

tochwill commented on src/app/lib/config/services/igbo.js:
```diff
@@ -178,6 +178,7 @@ export const service = {
         url: '/igbo/popular/read',
       },
     ],
+    navigationSection: 'Ngalaba',
```
We can move this into the `translations` object, and just call it 'section'.

AlistairGempf committed:
Add a comment about the hidden attribute

AlistairGempf committed:
Merge branch 'latest' into integrate-new-navigation

AlistairGempf committed:
Merge pull request #4952 from bbc/explain-hidden-attr-amp

Add a comment about the hidden attribute

AlistairGempf committed:
Merge branch 'latest' into integrate-new-navigation

Commit:
Rename variant to scriptLinkVariant. Make scriptLinkVariant the variant to be switched to.

Commit:
Merge branch 'integrate-new-navigation' of github.com:bbc/simorgh into 4009-use-brand-with-script-link

Commit:
Add setCookie() to ScriptLink.

Commit:
Add comments.

tochwill committed:
Update navigation unit tests

tochwill committed:
Merge branch 'integrate-new-navigation' into navigation-unit-tests

tochwill committed:
Update snapshots

tochwill committed:
Merge branch 'latest' into navigation-unit-tests

tochwill committed:
Merge branch 'latest' into integrate-new-navigation

tochwill committed:
Merge branch 'integrate-new-navigation' into navigation-unit-tests

Commit:
Fix setCookie() not working correctly.

Commit:
Merge branch 'integrate-new-navigation' of github.com:bbc/simorgh into 4009-use-brand-with-script-link

tochwill committed:
Refactor Nav container AMP constants

tochwill committed:
Update Nav Translations

tochwill committed:
update snapshot

Commit:
Add tests.

tochwill committed:
Update src/app/containers/Navigation/index.amp.test.jsx

Co-Authored-By: sadickisaac <48688870+sadickisaac@users.noreply.github.com>

Commit:
Remove setCookie(), use setPreferredVariantCookie(). Remove getPreferredVariantCookieName().

Commit:
Add skipLinkText translations. Update snapshots. Update tests.

Commit:
Merge branch 'latest' into integrate-new-navigation

Commit:
Merge branch 'integrate-new-navigation' of github.com:bbc/simorgh into 4009-use-brand-with-script-link

tochwill committed:
update bundle size config

tochwill committed:
Merge branch 'integrate-new-navigation' into update_nav_translations

Commit:
Update serbian, ukchina, zhongwen configs. Create getOtherVariant(), useCurrentVariant().

tochwill committed:
Merge pull request #4969 from bbc/update_nav_translations

Update nav translations

Commit:
Update renderScriptLink().

Commit:
Get setPreferredVariantCookie from UserContext.

tochwill committed:
Merge branch 'latest' into integrate-new-navigation

tochwill committed:
Merge branch 'integrate-new-navigation' into navigation-unit-tests

tochwill committed:
Merge branch 'navigation-unit-tests' of github.com:bbc/simorgh into navigation-unit-tests

tochwill committed:
Implement code review feedback

Commit:
Update tests. Remove useParams() hook.

Commit:
Merge branch 'integrate-new-navigation' of github.com:bbc/simorgh into 4009-use-brand-with-script-link

Commit:
Add tests.

tochwill committed:
Merge pull request #4968 from bbc/navigation-unit-tests

Navigation: add unit tests and refactor AMP constants

Commit:
Apply suggestions from code review.

Co-Authored-By: sadickisaac <48688870+sadickisaac@users.noreply.github.com>

Commit:
Apply suggestions from code review

Co-Authored-By: sadickisaac <48688870+sadickisaac@users.noreply.github.com>

Commit:
Merge branch '4009-use-brand-with-script-link' of github.com:bbc/simorgh into 4009-use-brand-with-script-link

Commit:
Merge branch 'integrate-new-navigation' of github.com:bbc/simorgh into 4009-use-brand-with-script-link

tochwill committed:
Cypress test for AMP navigation toggling

tochwill committed:
Rename amp toggling test

Commit:
Update serbian, ukchina, zhongwen configs. Update scriptLink switching functionality.
- Update ServiceContextProvider to include prefferedServiceVariant and otherVariant.
- Update withContext to update scriptLink to opposite variant.

Commit:
Update test.

tochwill committed:
Add clarity to services tested

Commit:
Use RequestContext.

Commit:
Undo .env changes.

Commit:
Merge branch '4009-use-brand-with-script-link' of github.com:bbc/simorgh into 4009-use-brand-with-script-link

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into 4009-use-brand-with-script-link

Commit:
Update test.

Commit:
Merge branch 'latest' into integrate-new-navigation

Commit:
Revert "Use RequestContext."

This reverts commit 7630067afb8874ee5068737b2dc717275437f288.

Commit:
Update renderScriptLink

Commit:
Merge branch 'integrate-new-navigation' of github.com:bbc/simorgh into 4009-use-brand-with-script-link

Commit:
Update variant configs.

Commit:
Update bundleSizeConfig.

Commit:
Update e2es.

tochwill committed:
Merge pull request #4990 from bbc/amp-nav-toggle-test

Cypress test for AMP navigation toggling

Commit:
Merge branch 'integrate-new-navigation' of github.com:bbc/simorgh into 4009-use-brand-with-script-link

Commit:
Update snapshots.

tochwill committed:
Implement nav animation

tochwill committed:
Merge branch 'latest' into integrate-new-navigation

Commit:
Revert scriptLink config editing.

Commit:
Remove reassignment.

Commit:
Merge branch 'integrate-new-navigation' of github.com:bbc/simorgh into 4009-use-brand-with-script-link

tochwill committed:
Merge branch 'latest' into integrate-new-navigation

tochwill committed:
Merge branch 'integrate-new-navigation' into nav-animation

tochwill committed:
Fix navigation tests

Commit:
Merge branch 'integrate-new-navigation' of github.com:bbc/simorgh into 4009-use-brand-with-script-link

Commit:
Merge branch 'latest' into integrate-new-navigation

Commit:
Merge branch 'integrate-new-navigation' of github.com:bbc/simorgh into 4009-use-brand-with-script-link

AlistairGempf committed:
Use the service name from the e2e test config

AlistairGempf committed:
Merge branch 'latest' of https://github.com/bbc/simorgh into integrate-new-navigation

AlistairGempf committed:
Merge branch 'integrate-new-navigation' of https://github.com/bbc/simorgh into e2e-test-service-name

OlgaLyubin committed:
Bump up navigation version

OlgaLyubin committed:
Resolve conflicts and update snaps

AlistairGempf committed:
Merge pull request #5051 from bbc/e2e-test-service-name

Use service name in e2e

OlgaLyubin committed:
Merge branch 'integrate-new-navigation' of github.com:bbc/simorgh into nav-animation

OlgaLyubin committed:
Merge branch 'latest' into integrate-new-navigation

AlistairGempf committed:
Merge branch 'latest' of https://github.com/bbc/simorgh into integrate-new-navigation

tochwill committed:
Merge branch 'integrate-new-navigation' into 4009-use-brand-with-script-link

tochwill committed:
Update snapshots

tochwill committed:
Merge pull request #4948 from bbc/4009-use-brand-with-script-link

Use Brand with  Script Link component on services with variants

OlgaLyubin committed:
Resolve conflicts

OlgaLyubin committed:
Update test

OlgaLyubin committed:
Update bundle size

OlgaLyubin committed:
Pull in new navigation version

OlgaLyubin committed:
Merge pull request #5057 from bbc/nav-animation

Integrate canonical navigation animation

OlgaLyubin committed:
Resolve merge conflicts

OlgaLyubin committed:
Resolve conflicts

OlgaLyubin committed:
Update to latest

OlgaLyubin committed:
Update to feature branch

OlgaLyubin committed:
Change scrollable nav to match psammead component

OlgaLyubin committed:
Delete media query hook

OlgaLyubin committed:
Update snapshot

OlgaLyubin committed:
Variable renaming

EinsteinNjoroge committed:
Merge branch 'latest' into integrate-new-navigation

OlgaLyubin committed:
Merge pull request #5087 from bbc/aria-navigation

Integrate changes for scrollable navigation

tochwill committed:
Add script switch and mobile nav e2es

tochwill committed:
Merge branch 'latest' into integrate-new-navigation

tochwill committed:
Merge branch 'integrate-new-navigation' into e2e-tests-nav

OlgaLyubin committed:
Pull in newest navigation and brand versions

OlgaLyubin committed:
Update snapshots

tochwill committed:
Update script link version

tochwill committed:
Update Snapshots

tochwill committed:
Restore media query hook, and ensure nav remains visible when screen width changed

tochwill committed:
Merge pull request #5163 from bbc/fix-hidden-navigation

Ensure nav remains visible when screen width changes

tochwill committed:
Update script link version

tochwill committed:
Unit test for navigation links

tochwill committed:
Merge branch 'integrate-new-navigation' into e2e-tests-nav

tochwill committed:
Add script link test for default variant

tochwill committed:
Merge branch 'latest' into integrate-new-navigation

tochwill committed:
fix snapshots

tochwill committed:
Remove duplicate dependencies

tochwill committed:
Merge branch 'integrate-new-navigation' into e2e-tests-nav

tochwill committed:
update nav version

tochwill committed:
Merge pull request #5178 from bbc/update-nav-version

Update navigation version

tochwill committed:
Merge pull request #5157 from bbc/e2e-tests-nav

Add E2E test for script link and mobile menu

OlgaLyubin committed:
Merge branch 'latest' into integrate-new-navigation

tochwill committed:
Ignore failing a11y test

PriyaKR committed:
Merge pull request #5223 from bbc/update-nav-version

Ignore failing a11y test

tochwill committed:
Merge branch 'latest' into integrate-new-navigation

tochwill committed:
Use latest version of skip link in brand
